### ISSUE TITLE
how to declare self. mlb?

### ISSUE THREAD

**Issue by nisnab** [mratsim/Amazon_Forest_Computer_Vision#4]
I find your code really helpful. Thank you. I see a difference between your kaggle and github code.
In your github code, you declare self.mlb, you put (if I am not wrong) all the total labels in the 'classes' array, However, in your kaggle code, it isn't done and kept empty. What is the difference and and which one of them is correct method?

**Comment by mratsim** [comment]
Oops sorry I didn't watch this repo at all and missed the Github notification.

Indeed in this repo I do this:
https://github.com/mratsim/Amazon-Forest-Computer-Vision/blob/master/src/p_dataload.py#L26-L32

But I think you missed that I also do the same in the tutorial on Kaggle:
https://www.kaggle.com/mratsim/starting-kit-for-pytorch-deep-learning
```python
    def __init__(self, csv_path, img_path, img_ext, transform=None):
    
        tmp_df = pd.read_csv(csv_path)
        assert tmp_df['image_name'].apply(lambda x: os.path.isfile(img_path + x + img_ext)).all(), \
"Some images referenced in the CSV file were not found"
        
        self.mlb = MultiLabelBinarizer()
        self.img_path = img_path
        self.img_ext = img_ext
        self.transform = transform

        self.X_train = tmp_df['image_name']
        self.y_train = self.mlb.fit_transform(tmp_df['tags'].str.split()).astype(np.float32)
```

fit_transform will add the classes to the MultiLabelBinarizer
